### PR TITLE
[feat] ACK EMR on EKS Controller pattern and example

### PR DESCRIPTION
**Pull request by season1946**
### What does this PR do?
https://github.com/awslabs/data-on-eks/issues/51

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/awslabs/data-on-eks/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR to add a new Add-on for [Terraform EKS Blueprints](https://github.com/aws-ia/terraform-aws-eks-blueprints) repo (if applicable)
- [ ] Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: analytics/terraform/emr-eks-ack-crossplane/addons.tf [hunk 1]
```diff
@@ -19,19 +19,7 @@ module "eks_blueprints_kubernetes_addons" {
   #   Further tuning for CoreDNS is to leverage NodeLocal DNSCache -> https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/
   #---------------------------------------------------------
   enable_coredns_autoscaler = true
```
Comment (vara-bonthu) [inline analytics/terraform/emr-eks-ack-crossplane/addons.tf:21]: You can remove this as well. `coredns_autosclaer` is enabled by default in EKS Blueprints

File: analytics/terraform/emr-eks-ack-crossplane/addons.tf [hunk 2]
```diff
@@ -139,3 +127,22 @@ module "eks_blueprints_kubernetes_addons" {
 
   tags = local.tags
 }
+
+
```
Comment (vara-bonthu) [inline analytics/terraform/emr-eks-ack-crossplane/addons.tf:131]: remove extra line

File: analytics/terraform/emr-eks-ack-crossplane/addons.tf [hunk 3]
```diff
@@ -139,3 +127,22 @@ module "eks_blueprints_kubernetes_addons" {
...
+module "eks_ack_addons" {
+  source = "github.com/aws-ia/terraform-aws-eks-ack-addons"
+
+  cluster_id = module.eks_blueprints.eks_cluster_id
+
+  # Wait for data plane to be ready
+  data_plane_wait_arn = module.eks_blueprints.managed_node_group_arn[0]
+
+
```
Comment (vara-bonthu) [inline analytics/terraform/emr-eks-ack-crossplane/addons.tf:144]: remove extra line

File: analytics/terraform/emr-eks-ack-crossplane/data.tf [hunk 4]
```diff
@@ -10,6 +10,7 @@ data "aws_caller_identity" "current" {}
 
 data "aws_partition" "current" {}
 
+
```
Comment (vara-bonthu) [inline analytics/terraform/emr-eks-ack-crossplane/data.tf:13]: remove extra line

File: analytics/terraform/emr-eks-karpenter/addons.tf [hunk 5]
```diff
@@ -71,7 +71,12 @@ module "eks_blueprints_kubernetes_addons" {
   # Cluster Autoscaler
   #---------------------------------------
   enable_karpenter = true
+  karpenter_helm_config = {
 
+    repository_username = data.aws_ecrpublic_authorization_token.token.user_name
+    repository_password = data.aws_ecrpublic_authorization_token.token.password
+
```
Comment (vara-bonthu) [inline analytics/terraform/emr-eks-karpenter/addons.tf:78]: remove extra line

File: analytics/terraform/emr-eks-karpenter/data.tf [hunk 6]
```diff
@@ -2,6 +2,9 @@ data "aws_eks_cluster_auth" "this" {
   name = module.eks_blueprints.eks_cluster_id
 }
 
+data "aws_ecrpublic_authorization_token" "token" {}
+
+
```
Comment (vara-bonthu) [inline analytics/terraform/emr-eks-karpenter/data.tf:7]: remove extra line

**Review by vara-bonthu — COMMENTED** [review]
Thanks for the PR @season1946 ! Few minor comments

File: analytics/terraform/emr-eks-ack-crossplane/addons.tf (hunk 1 above)
Comment (season1946) [reply analytics/terraform/emr-eks-ack-crossplane/addons.tf:21]: removed

**Review by vara-bonthu — APPROVED** [review]
Thanks @season1946 👍🏼

**Pull request merged**